src/devices/Tlc59711/Tlc59711.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Iot.Device.Spi;
+
+namespace Iot.Device.Tlc59711
+{
+    /// <summary>
+    /// On-board LED on the device.
+    /// </summary>
+    public class Tlc59711 : IDisposable
+    {
+        private const int ChannelsPerDriver = 12;
+
+        private readonly ushort[] _pwmBuffer;
+        private readonly int _numberOfDrivers;
+        private SpiDevice _spiDevice;
+        private byte _brightnessRed;
+        private byte _brightnessGreen;
+        private byte _brightnessBlue;
+        private byte[] _dataToWrite;
+
+        /// <summary>
+        /// Creates a new instance of the Tlc59711.
+        /// </summary>
+        public Tlc59711(SpiDevice spiDevice, int numberOfDrivers = 1)
+        {
+            _numberOfDrivers = numberOfDrivers;
+
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+
+            // default 100% brigthness
+            _brightnessRed = _brightnessGreen = _brightnessBlue = 0x7f;
+
+            // 12 channels per driver
+            _pwmBuffer = new ushort[numberOfDrivers * ChannelsPerDriver];
+
+            // Two bytes per channel, plus 1 byte for the command, and 3 bytes for the global brightness control
+            _dataToWrite = new byte[_numberOfDrivers * (ChannelsPerDriver * 2 + 4)];
+        }
+
+        /// <summary>
+        /// Disposes SpiDevice instances
+        /// </summary>
+        public void Dispose()
+        {
+            _spiDevice?.Dispose();
+            _spiDevice = null;
+        }
+
+        /// <summary>
+        /// Call this to set the PWM level for a channel.
+        /// </summary>
+        /// <param name="channel">Channel, 0-based</param>
+        /// <param name="pwm">PWM level (0 = minimum, 65535 = maximum)</param>
+        public void SetPWM(int channel, ushort pwm)
```

Comment:
Please remove this API if this is not suitable for regular PWM usage, alternative please create a way to create PwmChannel